crates/distribution-types/src/resolution.rs

```diff
@@ -70,6 +70,28 @@ impl Resolution {
     pub fn diagnostics(&self) -> &[ResolutionDiagnostic] {
         &self.diagnostics
     }
+
+    /// Filter the resolution to only include packages that match the given predicate.
+    pub fn filter(&self, predicate: impl Fn(&ResolvedDist) -> bool) -> Self {
```

Comment:
nit: Should this take `self` instead of `&self`?

Reply:
Done